### DISCUSSION TITLE
rheoMultiRegionFoam - using coupledT boundary condition

### DISCUSSION THREAD

**Discussion by am2021-create** [fppimenta/rheoTool#58] [Q&A]
Hello  Francisco

May you kindly assist me, I am simulating a conjugate heat transfer in concentric pipes. With the following regions inner pipe, solid wall (separating the two  fluids) and outer pipe. 

So I have created a single global blockMesh file with the below boundaries. 
defaultPatch
{
    name walls;
    type wall;
} 
boundary
(

    inlet_C
    {
        type patch;
        faces
        (
            (0 1 2 3)
            (2 3 8 9)
            (8 9 13 12)
            (13 12 0 1)
            (0 3 8 12)
        );
    }

    outlet_C
    {
        type patch;
        faces
        (
            (4 5 6 7)
            (6 7 10 11)
            (15 11 10 14)
            (15 14 4 5)
            (4 7 10 14)
        );
    }

inlet_S
    {
        type patch;
        faces
        (
            (20 28 29 21)
            (29 21 22 30)
            (31 30 22 23)
            (31 23 20 28)
        );
    }

    outlet_S
    {
        type patch;
        faces
        (
            (16 24 25 17)
            (25 17 18 26)
            (18 26 27 19)
            (27 19 16 24)
        );
    }

   outerpipe
    {
        type wall;
        faces
        (
            (24 28 29 25)
            (25 29 30 26)
            (26 30 31 27)
            (31 27 28 24)
        );
    }
And coupled patches were automatically created when running the application **splitMeshRegions -cellZones -overwrite**, these are solidwall-to-innerFluid, solidwall-to-outerFluid, innerFluid-to-solidwall and outerFluid-to-solidwall. 
I tried using the **coupledT boundary condition** in the 0-directly file at these boundaries ( coupled patches), without creating the createPatchDict . And I get the below error

Running blockMesh on /home/machine/OpenFOAM/machine-7/run/heatExchanger
Running splitMeshRegions on /home/machine/OpenFOAM/machine-7/run/heatExchanger
Running rheoMultiRegionFoam on /home/machine/OpenFOAM/machine-7/run/heatExchanger
Running paraFoam on /home/machine/OpenFOAM/machine-7/run/heatExchanger
cp: cannot stat './200': No such file or directory
cp: cannot create regular file './runCoupledT/200/innerFluid/T': No such file or directory
cp: cannot create regular file './runCoupledT/200/outerFluid/T': No such file or directory
cp: cannot create regular file './runCoupledT/200/solidWall/T': No such file or directory

I was also unable to added coupled patches in the mesh file where I could have used this ( taken from the rheoMultiRegionFoam/Slabs tutorial)
inter1
  {
    type regionCoupledAMI;
    neighbourPatch  inter0;
    neighbourRegion solid0;
    faces
    (
      (6 3 11 14)           
    );

My question is in rheoTool do I always need to build individual meshes for each region so as to able to use the coupledT boundary condition at coupled patches which comes with creating the createPatchDict first. Is there a way around this when using a single global mesh

**Comment by fppimenta** [comment]
For these simulations you always need different regions and coupling patches between regions. The way you do it doesn't matter and there are several ones.

**Comment by am2021-create** [comment]
Thank you for your reply,

I have created different regions in the global mesh file using
splitMeshRegions -cellZones overwrite. But I did not create a mesh for each
individual region in addition to the global mesh.

I tried to use the boundary condition, and the error that I got is that:
innerfluid-to-solidwall region is not regionCoupledAMI

Kindly advice/ suggest on any one  particle way I can use the boundary
condition which out having to create the createPatchDict

**Comment by fppimenta** [comment]
If the whole mesh is amenable with blockmesh, simply follow what is done in the rheoMultiRegionFoam/slabs tutorial. I am sure concentric cylinders can be built in that simple way.

**Comment by am2021-create** [comment]
hello, thank you for your response. I was unable to do it similar to the rheoMultiRegionFoam/slabs tutorial. BlockMesh gave me the below error
Creating single patch per inter-region interface.

--> FOAM FATAL ERROR: 
Cell 216000 with cell centre (-0.0170361 0.018429 0.0208333) is multiple zones. This is not allowed.
It is in zone innerFluid and in zone solidwall

    From function void getZoneID(const Foam::polyMesh&, const cellZoneMesh&, Foam::labelList&, Foam::labelList&)
    in file splitMeshRegions.C at line 1172.

FOAM exiting

**Comment by fppimenta** [comment]
The error doesn't come from blockMesh, but from splitMesh. Before you do the mesh for your case you need to understand what is being done in the tutorial. Keep trying, it will work.

**Comment by am2021-create** [comment]
oh okay, thanks. Will work on it
